### PR TITLE
fix(www): prevent dark flash when light mode

### PR DESCRIPTION
**Pull request by yangshun**
## What kind of change does this PR introduce?

Prevents the page from flashing from dark to light if the user selected a light theme (via the footer).

## What is the current behavior?

https://user-images.githubusercontent.com/1315101/152717469-6066bcf2-ce6a-431e-9f1b-06d6dbaf07f6.mp4

See that when the page first loads, the page is black and then it turns white.

## What is the new behavior?

By setting the final class name while the HTML is loading and before the browser first paints to the screen (via `<script>` tags), we can prevent the browser from rendering a version which doesn't match the user's preferences. This technique is being used in [Docusaurus v2](https://github.com/facebook/docusaurus/blob/aa446b7a9c0076e848708e9ea71b8c2e26698be0/packages/docusaurus-theme-classic/src/index.ts).

Also refactored some of the code to be more robust.

### CODE REVIEW

File: www/components/Layouts/Default.tsx [hunk 1]
```diff
@@ -8,25 +8,36 @@ type Props = {
...
-    }
-  }, [])
+  const initialMode =
+    typeof window != 'undefined' ? window.localStorage.getItem(MODE_STORAGE_KEY) === 'true' : true
+  const [darkMode, setDarkMode] = useState<boolean>(initialMode)
 
   const updateTheme = (isDarkMode: boolean) => {
-    document.documentElement.className = isDarkMode ? 'dark' : ''
+    document.documentElement.className = isDarkMode ? DARK_COLOR_SCHEME_CLASSNAME : ''
+    window.localStorage.setItem(MODE_STORAGE_KEY, (!darkMode).toString())
```
Comment (thebengeu) [inline www/components/Layouts/Default.tsx:34]: Should `(!darkMode)` be `isDarkMode` instead?

**Review by thebengeu — CHANGES_REQUESTED** [review]

**Comment by thebengeu** [comment]
I can't figure out why just from looking at the code, but if you refresh when in light mode, the toggle is set to dark mode:

Screenshot 2022-02-18 at 21 18 19

**Comment by thebengeu** [comment]
The default when there's nothing set in `localStorage` is now light mode, I don't think that's intended?

**Comment by MildTomato** [comment]
@yangshun this ready for another review?

**Comment by yangshun** [comment]
I don't think so, I'll fix it up soon and test it more thoroughly. Thanks
for the reminder!

Regards,
Yangshun

**Comment by MildTomato** [comment]
hey @yangshun, i'm just going to close this pull request for now. feel free to reopen it if you want to work on it again.

**Pull request closed without merging**